jkube-kit/resource/helm/src/test/java/org/eclipse/jkube/kit/resource/helm/HelmRepoTypeTest.java

```diff
@@ -93,12 +94,12 @@ public void createConnection_withArtifactoryAndNoAuth_shouldReturnConnection() t
   }
 
   @Test
-  public void createConnection_withNexusAndNoAuthAndTarGzExtension_shouldReturnConnectionToTgzUrl() throws IOException {
+  void createConnection_withNexusAndNoAuthAndTarGzExtension_shouldReturnConnectionToTgzUrl() throws IOException {
     // Given
     helmRepositoryBuilder.url("https://example.com");
     // When
     final HttpURLConnection result = HelmRepository.HelmRepoType.NEXUS
-        .createConnection(temporaryFolder.newFile("chart.tar.gz"), helmRepositoryBuilder.build());
+        .createConnection(new File(temporaryFolder, "chart.tar.gz"), helmRepositoryBuilder.build());
```

Comment:
Why didn't you use File.createTempFile in this case?

Reply:
I tried but it gives
```java
java.lang.AssertionError: 
Expecting
  <sun.net.www.protocol.https.DelegateHttpsURLConnection:https://example.com/chart1833669191392288151tar.gz>
to have a property or a field named <"url"> with value
  <https://example.com/chart.tgz>
but value was:
  <https://example.com/chart1833669191392288151tar.gz>
```

Reply:
because your temp file has a wrong extension. The resulting filename should be `chart1833669191392288151.tar.gz` instead of `chart1833669191392288151tar.gz`

I understand that the temp file suffix should be  `.tar.gz` and that you were using `tar.gz`

Reply:
This time I used `.tar.gz` still it is not working
```java
java.lang.AssertionError: 
Expecting
  <sun.net.www.protocol.https.DelegateHttpsURLConnection:https://example.com/chart18279381371330993283.tgz>
to have a property or a field named <"url"> with value
  <https://example.com/chart.tgz>
but value was:
  <https://example.com/chart18279381371330993283.tgz>
```

Reply:
According to [Java Docs](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/File.html#createTempFile(java.lang.String,java.lang.String,java.io.File)) `name of the new file will be generated by concatenating the prefix, five or more internally-generated characters, and the suffix.`

which means `File.createTempFile(...)` will not work because the resulting filename should be `chart.tgz` instead of `chart18279381371330993283.tgz`

but `Files.createFile(temporaryFolder.toPath().resolve("chart.tgz")).toFile()` is working because it creates an actual file with name `chart.tgz` instead of `chart18279381371330993283.tgz`

Reply:
And [JUnit4](https://junit.org/junit4/javadoc/4.12/src-html/org/junit/rules/TemporaryFolder.html#line.65) creates a new file using `new File(getRoot(), filename)`

Reply:
> And [JUnit4](https://junit.org/junit4/javadoc/4.12/src-html/org/junit/rules/TemporaryFolder.html#line.65) creates a new file using new File(getRoot(), filename) that's why I used new File(temporaryFolder, "chart.tar.gz")

JUnit 4 creates a new file by using `TemporaryFolder#newFile`, which **is not** the same as `new File(...)`.

If the filename needs to be `chart.tgz`, then by all means do a `Files#createFile` with the appropriate filename (which again, **is not** the same as new File(...)).

In case it's not clear, `new File(...)` won't create a file or a directory, it's just a reference to a path in your filesystem that might or might not exist.